### ISSUE TITLE
 The problem of a dataset being too small

### ISSUE THREAD

**Issue by zzz135790** [koc-lab/TrainableFrFT#3]
Sorry to bother you again. I want to know about an issue. Your trainable FRFT (Fractional Fourier Transform) indeed performs well, but on a relatively small dataset, it doesn't perform as well as the ordinary FFT (Fast Fourier Transform). I want to understand if this is because the dataset is too small, leading to insufficient learning of the parameters in the FRFT, and thus causing its performance to be inferior to that of the ordinary FFT.Thank you.

**Comment by kocemir** [comment]
Hello, it is unlikely that it performs worse than the ordinary FFT. Can you specify the task and dataset in detailed?

**Comment by zzz135790** [comment]
Hello, the dataset is not convenient to provide, it is radio signals, and the task is to identify which radio station a signal segment belongs to. Of course, it performs better than FFT on most datasets, but it does not perform as well as FFT on this relatively small dataset (I am using two-dimensional FFT here). Also, I wanted to ask if Trainable FRFT has been attempted in two-dimensional versions. I believe that in some cases, two-dimensional FRFT may perform better.

**Comment by kocemir** [comment]
In our source codes, we also provide the transformations for 2D signals for both dimension. Basically, frft(frft(X2D,dim=-1),dim=-2) must give you the 2D FRFT of the signal.

**Comment by tunakasif** [comment]
> In our source codes, we also provide the transformations for 2D signals for both dimension. Basically, frft(frft(X2D,dim=-1),dim=-2) must give you the 2D FRFT of the signal.

As described by @kocemir, we have the flexibility to apply transformation along any given dimension. We provide the `torch-frft` as a standalone installable Python package on its own [GitHub page](https://github.com/tunakasif/torch-frft), which has further documentation for the possible use cases in its `README.md`. A minimal example could be the following:

```python
import torch
from torch_frft.frft_module import frft
from torch_frft.dfrft_module import dfrft

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

N = 128
a0, a1 = 1.25, 0.75
X = torch.rand(N, N, device=device)

# 2D FRFT
Y1 = frft(frft(X, a0, dim=0), a1, dim=1)

# 2D DFRFT
Y2 = dfrft(dfrft(X, a0, dim=0), a1, dim=1)
```

Additionally, for your initial question, sometimes ordinary DFT ($a=1$) may be the optimal transformation for a given dataset. We provide a neural network layer with the fractional Fourier transform where the transform order is trainable. The resulting order and the overall performance of your model are the results of the input data and the remaining architecture.

**Comment by zzz135790** [comment]
Thank you for your patient responses. This has been a great help to a beginner like me. I wish you success in your research and happiness in life.
